runtime/src/bank.rs

```diff
@@ -1155,16 +1155,23 @@ struct VoteWithStakeDelegations {
     delegations: Vec<(Pubkey, StakeAccount<()>)>,
 }
 
+type VoteWithStakeDelegationsMap = DashMap<Pubkey, VoteWithStakeDelegations>;
```

Comment:
My preference:
Break this pr up. do your remans and types in 1 pr.
Break up the fns in a different pr.
I may then do the change of the stakes cache behavior in a third pr that just focuses on that. There are several things going on here in the same pr, which makes me nervous.